### PR TITLE
Tweaks to makefile for easier setup

### PR DESCRIPTION
**Pull request by tobyjoe**
This PR adds a setup step to the default make target (in case bundle isn't installed, for example), and makes it possible to provide variables to switch AWS profiles and stack JSON publication bucket endpoints.

It also adds a target to update your stack.

I did remove the date stamping of stacks because it made updates difficult. We could cache the value locally or something... or make it a parameter?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,30 +16,40 @@ build/aws-stack.json: $(wildcard templates/*.yml)
 	bundle exec cfoo $^ > $@
 
 setup:
-	which bundle || gem install bundler --no-ri --no-rdoc
-	bundle install --path vendor/bundle
+	bundle check || ((which bundle || gem install bundler --no-ri --no-rdoc) && bundle install --path vendor/bundle)
+	cp config.json.example config.json
```
Comment (lox) [inline Makefile]: Not sure this is wise, perhaps it could just fail if config.json doesn't exist with a helpful message?

**Comment by lox** [comment]
Thanks, reviewing.

Might I suggest https://github.com/99designs/aws-vault for profile management? :)

File: Makefile [hunk 2]
```diff
@@ -14,30 +16,40 @@ build/aws-stack.json: $(wildcard templates/*.yml)
...
 	aws cloudformation create-stack \
 	--output text \
-	--stack-name buildkite-$(shell date +%Y-%m-%d-%H-%M) \
+	--stack-name buildkite \
 	--disable-rollback \
 	--template-body "file://${PWD}/build/aws-stack.json" \
 	--capabilities CAPABILITY_IAM \
-	--parameters '$(shell cat config.json)'
+	--parameters '$(shell cat config.json)' \
+	--profile "$(AWS_PROFILE)"
```
Comment (lox) [inline Makefile]: AFAIK, the aws-cli will read the AWS_PROFILE env anyway, so this shouldn't be needed.

File: Makefile (hunk 1 above)
Comment (tobyjoe) [inline Makefile]: Yeah, I think you're right. A nice "run this command and edit your conf!" message is a better idea. I'll patch that.

File: Makefile (hunk 2 above)
Comment (tobyjoe) [inline Makefile]: You're totally spot-on with that "AFAIK" - I'll tear those bits out! (does explain why my tests were FLAWLESS, though! heh)

**Comment by tobyjoe** [comment]
Updated the branch in my fork with feedback. 😍

**Comment by lox** [comment]
Looking good, I think with the new update changelist stuff in cloudformation updating the stack makes more sense.

File: README.md [hunk 3]
```diff
@@ -21,6 +21,17 @@ aws cloudformation create-stack \
   --parameters <(cat config.json)
 ```
 
+
+Alternately, if you prefer to use this repo, clone it and run the following command to set up things locally and create a remote stack.
+```bash
+make && make create-stack
```
Comment (lox) [inline README.md]: This is redundant, just `make create-stack` will do it. By itsself `make` just uses the first target, which is build, which ends up being a dependency of `create-stack` anyway.

File: README.md [hunk 4]
```diff
@@ -21,6 +21,17 @@ aws cloudformation create-stack \
...
+
+# If you need to specify a profile:
```
Comment (lox) [inline README.md]: I'd be tempted to just replace this with something like:

"You can use any of the `AWS...` environment variables that the aws-cli supports."

File: Makefile [hunk 5]
```diff
@@ -14,30 +15,41 @@ build/aws-stack.json: $(wildcard templates/*.yml)
...
+
 	aws cloudformation create-stack \
 	--output text \
-	--stack-name buildkite-$(shell date +%Y-%m-%d-%H-%M) \
+	--stack-name buildkite \
 	--disable-rollback \
 	--template-body "file://${PWD}/build/aws-stack.json" \
 	--capabilities CAPABILITY_IAM \
-	--parameters '$(shell cat config.json)'
+	--parameters '$(shell cat config.json)' \
```
Comment (lox) [inline Makefile]: Trailing back slash? On the update too.

File: Makefile [hunk 6]
```diff
@@ -14,30 +15,41 @@ build/aws-stack.json: $(wildcard templates/*.yml)
...
 
 build-ami:
 	cd packer/; packer build buildkite-ami.json
 
 upload: build/aws-stack.json
-	aws s3 sync --acl public-read build s3://buildkite-aws-stack/
+	aws s3 sync --acl public-read build s3://${BUILDKITE_STACK_BUCKET}/
 
 create-stack: templates/mappings.yml build/aws-stack.json
+	test -s config.json || { echo "Please create a config.json file"; exit 1; }
```
Comment (lox) [inline Makefile]: Could this just be its own target?

**Comment by lox** [comment]
Thank you @tobyjoe

**Pull request merged**